### ISSUE TITLE
Improve the PPOLearner to support continuous action space

### ISSUE THREAD

**Issue by findmyway** [JuliaReinforcementLearning/ReinforcementLearningZoo.jl#92]
Necessary changes:

1. Add a field of `dist` into the `PPOLearner` (just like @norci  did in [VPG](https://github.com/JuliaReinforcementLearning/ReinforcementLearningZoo.jl/blob/34aea9030577ed5cea6630ad02dc60e6c4cfd9a6/src/algorithms/policy_gradient/ppo.jl#L31) )

1. Following method needs to be extended to recognize environments with continuous action space. Currently the PPOLearner is assumed to return a (batch of ) logits. I'd suggest to rename the `PPOLearner` into `PPOPolicy` and return an action directly. 
https://github.com/JuliaReinforcementLearning/ReinforcementLearningZoo.jl/blob/34aea9030577ed5cea6630ad02dc60e6c4cfd9a6/src/algorithms/policy_gradient/ppo.jl#L77

1. [GausianNetwork](https://github.com/JuliaReinforcementLearning/ReinforcementLearningZoo.jl/pull/87/files#diff-dc9b8ba63840800093295c526ec3aa91R9-R13) is also needed.

1. Calculating entropy loss in `update!` is hard coded. Better to split it into a function to support continuous distribution (or reuse the one in StatsBase or Distributions. But use them with caution! I had some problems using them with Zygote before)

**Comment by norci** [comment]
I think tianshou's implementation is a good reference:

https://github.com/thu-ml/tianshou/blob/master/test/continuous/test_ppo.py
https://github.com/thu-ml/tianshou/blob/master/tianshou/utils/net/continuous.py

**Comment by norci** [comment]
I found a few papers about continuous control. they did not use Gaussian policy.

* http://proceedings.mlr.press/v70/chou17a.html
* https://deepai.org/publication/continuous-control-with-deep-reinforcement-learning
* https://deepai.org/publication/learning-continuous-control-policies-by-stochastic-value-gradients
* https://arxiv.org/abs/1802.07564

I'm still trying to understand these papers.

**Comment by stasti** [comment]
i think to have Gaussian policy is useful at least as the baseline before jumping to more advanced methods. and it is still used in many works.
